effekt/shared/src/main/scala/effekt/symbols/Symbol.scala

```diff
@@ -38,7 +38,7 @@ trait Symbol {
     case _             => false
   }
 
-  def show: String = name.toString + id
+  def show: String = s"${name}_${id}"
```

Comment:
This is somewhat necessary, some identifiers from pattern matching have `name := "v_r_11029323"` already, so if we were to just append the `id := 42`, we'd get a much longer number like `v_r_1102932342`.
Instead, we now get `v_r_11029323_42`.
The cost is that now instead of having `Char342` and `List45`, we have `Char_342` and `List_45`...

Alternative designs:
1. choose a different symbol to append like `@` (since it cannot appear in an identifier anyways) and then sanitise for backends
2. change how all of those `v_r_...` get generated.

Reply:
I'll try to do 2., since I think it's viable.

Reply:
I think having this distinction explicitly (always counting on `${name}_${id}`) is helpful, so I'll keep it :)

Reply:
I would keep the `_`. I don't think it adds too much clutter, and it *guarantees* that we do not clash with some existing name (e.g. `Int12` and `Int1` renamed to `Int12` by this or sth), even if those are unlikely.